opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -216,136 +257,143 @@ impl<R: RuntimeChannel> LogProcessor for BatchLogProcessor<R> {
             .try_send(BatchMessage::Shutdown(res_sender))
             .map_err(|err| LogError::Other(err.into()))?;
 
-        futures_executor::block_on(res_receiver)
-            .map_err(|err| LogError::Other(err.into()))
-            .and_then(std::convert::identity)
+        let (sender, receiver) = mpsc::sync_channel(1);
```

Comment:
there is a one shot::channel, and another try_send just above this. I guess they were leftovers and should be removed.